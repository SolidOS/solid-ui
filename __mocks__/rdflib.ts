export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    elements: [],
    doc: () => {
      return {
        equals: () => {},
        value: ''
      }
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {
}

export function literal() {
}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    match: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    bottomTypeURIs: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
  editable() {}
  update() {}
}
