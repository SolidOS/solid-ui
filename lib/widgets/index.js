"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  publicData: true
};
exports.publicData = void 0;
var _peoplePicker = require("./peoplePicker");
Object.keys(_peoplePicker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _peoplePicker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _peoplePicker[key];
    }
  });
});
var _dragAndDrop = require("./dragAndDrop");
Object.keys(_dragAndDrop).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _dragAndDrop[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _dragAndDrop[key];
    }
  });
});
var _buttons = require("./buttons");
Object.keys(_buttons).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _buttons[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _buttons[key];
    }
  });
});
var _iconLinks = require("./buttons/iconLinks");
Object.keys(_iconLinks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _iconLinks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _iconLinks[key];
    }
  });
});
var _error = require("./error");
Object.keys(_error).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _error[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _error[key];
    }
  });
});
var _forms = require("./forms");
Object.keys(_forms).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _forms[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _forms[key];
    }
  });
});
var _autocompleteBar = require("./forms/autocomplete/autocompleteBar");
Object.keys(_autocompleteBar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _autocompleteBar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _autocompleteBar[key];
    }
  });
});
var _autocompletePicker = require("./forms/autocomplete/autocompletePicker");
Object.keys(_autocompletePicker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _autocompletePicker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _autocompletePicker[key];
    }
  });
});
var _publicData = _interopRequireWildcard(require("./forms/autocomplete/publicData"));
exports.publicData = _publicData;
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
//# sourceMappingURL=index.js.map