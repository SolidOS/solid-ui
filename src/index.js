/*
The MIT License (MIT)

Copyright (c) 2015-2016 Solid

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.


If you would like to know more about the solid Solid project, please see
https://github.com/solid/solid
*/
'use strict'
/**
 * Provides a Solid client helper object (which exposes various static modules).
 * @module solidUi.js
 * @main solidUi.js
 */

/**
 * @class SolidUi
 * @static
 */

var rdf = require('rdflib') // pull in first avoid cross-refs
var ns = require('./ns')

var UI = {
  acl: require('./acl'),
  aclControl: require('./acl-control'),
  icons: require('./iconBase'),
  log: require('./log'),
  matrix: require('./matrix'),
  messageArea: require('./messageArea'),
  ns: ns,
  pad: require('./pad'),
  rdf:  rdf,
  signin: require('./signin'),
  solid: require('solid-client'),
  store: require('./store'),
  table: require('./table'),
  tabs: require('./tabs'),
  utils: require('./utils'),
  widgets: require('./widgets')
}

module.exports = UI
