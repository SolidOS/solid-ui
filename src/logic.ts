// This module of solid-ui has a main quadstore for the app to use
//

import * as debug from './debug'
import { fetch } from 'solid-auth-client'
// import { SolidLogic } from 'solid-logic'
import { SolidLogic } from 'solid-logic'

export const solidLogicSingleton = new SolidLogic({ fetch })

debug.log('Unique quadstore initialized.')

// ends
