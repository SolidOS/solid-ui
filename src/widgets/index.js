/**
 * General purpose utility functions used in the panes
 * oshani@csail.mit.edu
 *
 * Includes form-oriented widgets  timbl@w3.org
 *
 * sign-in sign-up widgets are in signin.js
 *
 *  Note... For pointers to posssible text-editing code, see
 *  http://stackoverflow.com/questions/6756407/what-contenteditable-editors
 */

import * as debug from '../debug'

// var aclModule = require('./acl.js')

// Each widget should ideally live in its own file.  In order to break up this
// monolithic widget index over time, we should add new widgets to the
// 'lib/widgets/' directory, and re-export them like so:
//
// (In order to avoid name collisions, it is safely assumed that modules don't
// export widgets with the same name)
var widgets = Object.assign(
  {},
  require('./peoplePicker'),
  require('./dragAndDrop'), // uploadFiles etc
  require('./error'), // UI.widgets.errorMessageBlock
  require('./buttons'),
  require('./forms')
)

module.exports = widgets
debug.log('@@@ widgets: ', widgets)

// ends
