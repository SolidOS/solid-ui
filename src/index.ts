/*
The MIT License (MIT)

Copyright (c) 2015-2016 Solid

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

If you would like to know more about the solid Solid project, please see
https://github.com/solid/solid
*/
'use strict'
/**
 * Provides a Solid client helper object (which exposes various static modules).
 * @module solidUi.js
 * @main solidUi.js
 */

/**
 * @class SolidUi
 * @static
 */

// REMOVE @ts-ignore as you migrate files to TypeScript
import * as rdf from 'rdflib' // pull in first avoid cross-refs
// @ts-ignore
import ns from './ns'
import { acl, aclControl } from './acl/index'
import { authn } from './authn/index'
import { create } from './create/index'
// @ts-ignore
import icons from './iconBase'
import * as log from './log'
import { matrix } from './matrix/index'
import { media } from './media/index'
// @ts-ignore
import messageArea from './messageArea'
// @ts-ignore
import { infiniteMessageArea } from './chat/infinite'
// @ts-ignore
import * as pad from './pad'
// @ts-ignore
import * as participation from './participation'
// @ts-ignore
import preferences from './preferences'
// @ts-ignore
import { solidLogicSingleton } from './logic'
// @ts-ignore
import style from './style'
// @ts-ignore
import table from './table'
import * as tabs from './tabs'
// @ts-ignore
import utils from './utils'
import widgets from './widgets'
import versionInfo from './versionInfo'
import { initHeader } from './header'

const dom = window ? window.document : null // Idea that UI.dom can be adapted in non-browser environments
const store = solidLogicSingleton.store

if (typeof window !== 'undefined') {
  ;(<any>window).UI = {
    ns,
    rdf,
    acl,
    aclControl,
    authn,
    create,
    dom,
    icons,
    log,
    matrix,
    media,
    messageArea,
    infiniteMessageArea,
    pad,
    participation,
    preferences,
    solidLogicSingleton,
    store,
    style,
    table,
    tabs,
    utils,
    widgets,
    versionInfo,
    initHeader
  } // Simpler access by non-node scripts
}

export {
  ns,
  rdf,
  acl,
  aclControl,
  authn,
  create,
  dom,
  icons,
  log,
  matrix,
  media,
  messageArea,
  infiniteMessageArea,
  pad,
  participation,
  preferences,
  solidLogicSingleton,
  store,
  style,
  table,
  tabs,
  utils,
  widgets,
  versionInfo,
  initHeader
}
