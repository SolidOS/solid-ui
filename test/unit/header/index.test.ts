import { silenceDebugMessages } from '../../helpers/setup'
import {
  initHeader,
  createLoginSignUpButtons,
  createUserMenuButton,
  createUserMenuLink,
  createUserMenuItem,
  rebuildHeader,
  getProfileImg,
  createBanner,
  createUserMenu
} from '../../../src/header'
import { NamedNode } from 'rdflib'
// @ts-ignore
import widgets from '../../../src/widgets/index'
import { solidLogicSingleton } from '../../../src/logic'
// import { JSDOM } from 'jsdom'

const store = solidLogicSingleton.store

// SAM todos...
//  1. create specific tests to check for a class being assigned
silenceDebugMessages()
// const window = new JSDOM('<!DOCTYPE html><p>Hello world</p>').window
// const dom = window.document

describe('header', () => {
  // SAM not sure about this one
  const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
  it('is exposed on public API', async () => {
    expect(await initHeader(store, options)).toMatchSnapshot()
  })
})
describe('createLoginSignUpButtons', () => {
  // SAM needs a class check
  it('creates div', () => {
    expect(createLoginSignUpButtons()).toMatchSnapshot()
  })
  it('assigns class header-banner__login', () => {
    const signUpDiv = createLoginSignUpButtons()
    expect(signUpDiv.className).toContain('header-banner__login')
  })
})
describe('createUserMenuButton', () => {
  it('creates a button', () => {
    const label = 'testing button'
    const onclick = () => { window.alert('test') }
    expect(createUserMenuButton(label, onclick)).toMatchSnapshot()
  })
  it('checks to see if the class is header-user-menu__button', () => {
    const label = 'testing button'
    const onclick = () => { window.alert('test') }
    const button = createUserMenuButton(label, onclick)
    expect(button.className).toContain('header-user-menu__button')
  })
})

describe('createUserMenuLink', () => {
  it('creates a link', () => {
    const label = 'testing link'
    const url = 'http://www.test.com'
    expect(createUserMenuLink(label, url)).toMatchSnapshot()
  })
  it('assigns link class header-user-menu__link', () => {
    const label = 'testing link'
    const url = 'http://www.test.com'
    const menuLink = createUserMenuLink(label, url)
    expect(menuLink.className).toContain('header-user-menu__link')
  })
})
describe('createUserMenuItem', () => {
  it('creates a link', () => {
    const ulElement = document.createElement('ul')
    expect(createUserMenuItem(ulElement)).toMatchSnapshot()
  })
  it('assigns list item class header-user-menu__list-item', () => {
    const ulElement = document.createElement('ul')
    const listItem = createUserMenuItem(ulElement)
    expect(listItem.className).toContain('header-user-menu__list-item')
  })
})
describe('createUserMenu', () => {
  it('creates a menu....', () => {
    const pod = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')

    const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(createUserMenu(store, pod, options)).toMatchSnapshot()
  })
  it('creates the default options...', () => {
    const pod = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    // think of test here
    expect(createUserMenu(store, pod)).toMatchSnapshot()
  })
  it('assigns list item class header-banner__user-menu and header-user-menu', async () => {
    const pod = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    const userMenu = await createUserMenu(store, pod, options)
    expect(userMenu.className).toContain('header-banner__user-menu')
  })
})
// SAM look into this
describe('rebuildHeader', () => {
  it('creates a link', () => {
    const header = document.createElement('nav')
    const pod = new NamedNode('https://test.com')
    const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(rebuildHeader(header, store, pod, options)).toMatchSnapshot()
  })
})
describe('createBanner', () => {
  it('creates a link', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(await createBanner(store, pod, user, options)).toMatchSnapshot()
  })
  it('assigns list item class header-banner', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = { menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    const banner = await createBanner(store, pod, user, options)
    expect(banner.className).toContain('header-banner')
  })
  // SAM need to look into this further,
  it('check customized logo...', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = { logo: 'https://solidproject.org/assets/img/solid-emblem.svg', menuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(await createBanner(store, pod, user, options)).toMatchSnapshot()
    // check something else re logo expect(banner.className).toContain('header-banner')
  })
})

// getProfileImg, need to mock widgets.findImage(user)
describe('getProfileImg', () => {
  // SAM look into best way to get the store and what user to use..
  // this will also need a class check
  it.skip('returns an empty image', () => {
    // eslint-disable-next-line no-undef
    // eslint-disable-next-line no-undef
    const findImageSpy = jest.spyOn(widgets, 'findImage').mockImplementation(() => { return 'test' })
    const user = new NamedNode('http://test.com')
    expect(getProfileImg(store, user)).toMatchSnapshot()
    expect(findImageSpy).toBeCalled()
  })
})
