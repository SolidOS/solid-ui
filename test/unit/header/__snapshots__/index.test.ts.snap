// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBanner check customized logo... 1`] = `
<div
  class="header-banner-0-1-81"
>
  <a
    class="header-banner__link-0-1-68"
    href="https://test.com"
  >
    <img
      class="header-banner__icon-0-1-69"
      src="https://solidproject.org/assets/img/solid-emblem.svg"
    />
  </a>
  <div
    class="header-banner__user-menu-0-1-79 header-user-menu-0-1-80"
  >
    <button
      class="header-user-menu__trigger-0-1-78"
      type="button"
    >
      

      <svg
        fill="none"
        height="26"
        viewBox="0 0 26 26"
        width="26"
        xmlns="http://www.w3.org/2000/svg"
      >
        
    
        <path
          clip-rule="evenodd"
          d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
          fill="#D8D8D8"
          fill-rule="evenodd"
          stroke="#8B8B8B"
        />
        
    
        <mask
          height="26"
          id="mask0"
          mask-type="alpha"
          maskUnits="userSpaceOnUse"
          width="26"
          x="0"
          y="0"
        >
          
        
          <path
            clip-rule="evenodd"
            d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
            fill="white"
            fill-rule="evenodd"
            stroke="white"
          />
          
    
        </mask>
        
    
        <g
          mask="url(#mask0)"
        >
          
        
          <path
            clip-rule="evenodd"
            d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z"
            fill="#8B8B8B"
            fill-rule="evenodd"
          />
          
    
        </g>
        

      </svg>
    </button>
    <nav
      aria-hidden="true"
      class="header-user-menu__navigation-menu-0-1-77"
    >
      <ul
        class="header-user-menu__list-0-1-70"
      >
        <li
          class="header-user-menu__list-item-0-1-72"
        >
          <a
            class="header-user-menu__link-0-1-71"
            href="https://sharonstrats.inrupt.net/profile/card#me"
          />
        </li>
        <li
          class="header-user-menu__list-item-0-1-74"
        >
          <a
            class="header-user-menu__link-0-1-73"
            href="https://reflectechblog.wordpress.com/"
          />
        </li>
        <li
          class="header-user-menu__list-item-0-1-76"
        >
          <button
            class="header-user-menu__button-0-1-75"
          />
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`createBanner creates a link 1`] = `
<div
  class="header-banner-0-1-53"
>
  <a
    class="header-banner__link-0-1-40"
    href="https://test.com"
  >
    <img
      class="header-banner__icon-0-1-41"
      src="https://solidproject.org/assets/img/solid-emblem.svg"
    />
  </a>
  <div
    class="header-banner__user-menu-0-1-51 header-user-menu-0-1-52"
  >
    <button
      class="header-user-menu__trigger-0-1-50"
      type="button"
    >
      

      <svg
        fill="none"
        height="26"
        viewBox="0 0 26 26"
        width="26"
        xmlns="http://www.w3.org/2000/svg"
      >
        
    
        <path
          clip-rule="evenodd"
          d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
          fill="#D8D8D8"
          fill-rule="evenodd"
          stroke="#8B8B8B"
        />
        
    
        <mask
          height="26"
          id="mask0"
          mask-type="alpha"
          maskUnits="userSpaceOnUse"
          width="26"
          x="0"
          y="0"
        >
          
        
          <path
            clip-rule="evenodd"
            d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
            fill="white"
            fill-rule="evenodd"
            stroke="white"
          />
          
    
        </mask>
        
    
        <g
          mask="url(#mask0)"
        >
          
        
          <path
            clip-rule="evenodd"
            d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z"
            fill="#8B8B8B"
            fill-rule="evenodd"
          />
          
    
        </g>
        

      </svg>
    </button>
    <nav
      aria-hidden="true"
      class="header-user-menu__navigation-menu-0-1-49"
    >
      <ul
        class="header-user-menu__list-0-1-42"
      >
        <li
          class="header-user-menu__list-item-0-1-44"
        >
          <a
            class="header-user-menu__link-0-1-43"
            href="https://sharonstrats.inrupt.net/profile/card#me"
          />
        </li>
        <li
          class="header-user-menu__list-item-0-1-46"
        >
          <a
            class="header-user-menu__link-0-1-45"
            href="https://reflectechblog.wordpress.com/"
          />
        </li>
        <li
          class="header-user-menu__list-item-0-1-48"
        >
          <button
            class="header-user-menu__button-0-1-47"
          />
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`createLoginSignUpButtons creates div 1`] = `
<div
  class="header-banner__login-0-1-1"
>
  <div>
    <div
      class="SolidSignInOrSignUpBox"
      style="display: flex;"
    >
      <input
        style="padding: 1em; border-radius:0.5em; margin: 2em; font-size: 100%;background-color: #eef;"
        type="button"
        value="Log in"
      />
      <input
        style="padding: 1em; border-radius:0.5em; margin: 2em; font-size: 100%;background-color: #efe;"
        type="button"
        value="Sign Up for Solid"
      />
    </div>
  </div>
</div>
`;

exports[`createUserMenu creates a menu.... 1`] = `Promise {}`;

exports[`createUserMenu creates the default options... 1`] = `Promise {}`;

exports[`createUserMenuButton creates a button 1`] = `
<button
  class="header-user-menu__button-0-1-3"
/>
`;

exports[`createUserMenuItem creates a link 1`] = `
<li
  class="header-user-menu__list-item-0-1-7"
>
  <ul />
</li>
`;

exports[`createUserMenuLink creates a link 1`] = `
<a
  class="header-user-menu__link-0-1-5"
  href="http://www.test.com"
/>
`;

exports[`getProfileImg returns an empty image 1`] = `
"
<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"26\\" height=\\"26\\" viewBox=\\"0 0 26 26\\" fill=\\"none\\">
    <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z\\" fill=\\"#D8D8D8\\" stroke=\\"#8B8B8B\\"/>
    <mask id=\\"mask0\\" mask-type=\\"alpha\\" maskUnits=\\"userSpaceOnUse\\" x=\\"0\\" y=\\"0\\" width=\\"26\\" height=\\"26\\">
        <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z\\" fill=\\"white\\" stroke=\\"white\\"/>
    </mask>
    <g mask=\\"url(#mask0)\\">
        <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z\\" fill=\\"#8B8B8B\\"/>
    </g>
</svg>"
`;

exports[`header is exposed on public API 1`] = `undefined`;

exports[`rebuildHeader creates a link 1`] = `[Function]`;
